specs/Print.md

```diff
@@ -0,0 +1,473 @@
+
+You can use `CoreWebView2.PrintToPdfAsync` API to print a current page as a pdf file to disk and then use native printing APIs to print the document to a specific printer. However, this API requires the PDF to be printed to the disk which can be prohibitive at times for some financial customers if the printed information is sensitive and also printing to printer without a dialog requires additional effort on the developers for cross platform apps.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose following API's
+
+**Print API**: This API gives developers the ability to open a browser print preview dialog or system print dialog without requiring the end user interaction. This API consists of a Print method and can either open a browser print preview dialog or system print dialog based on COREWEBVIEW2_PRINT_DIALOG_KIND value.
+
+**Print without a print dialog**: This API gives developers the ability to print the current page without a print dialog to the specific printer. This API consists of an asynchronous PrintWithSettings method and a PrintSettings2 object.
+PrintSettings2 object extends PrintSettings to support printing to a printer.
+
+**Print to PDF Stream**: This API gives developers printable PDF data, which can be used as a preview in a custom print preview dialog. This API consists of an asynchronous PrintToPdfStream method and a PrintSettings2 object.
+
+
+# Examples
+## Print
+
+You can use the `Print` method to programmatically open a print dialog without end user interaction. Currently devs are using JavaScript `window.print()` using the `CoreWebView2.ExecuteScriptAsync` API to invoke the
+browser print preview dialog. But there isn't a way to invoke open the system print dialog programmatically nor will `window.print()` work if script is disabled on the page, or the document is a PDF or other non-HTML document.
+
+// This example shows the user a print dialog. If `printDialogKind` is COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER_PRINT_PREVIEW, opens a browser print preview dialog, otherwise opens a system print dialog.
+```cpp
+bool AppWindow::OpenPrintDialog(COREWEBVIEW2_PRINT_DIALOG_KIND printDialogKind)
+{
+     wil::com_ptr<ICoreWebView2_15> webView2_15;
+     CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+     CHECK_FEATURE_RETURN(webView2_15);
+     CHECK_FAILURE(webView2_15->Print(printDialogKind));
+     return true;
+}
+```
+
+```c#
+void OpenPrintDialog(object target, ExecutedRoutedEventArgs e)
+{
+     string printDialog = e.Parameter.ToString();
+     if (printDialog == "PrintPreview")
+     {
+         // Opens the browser print preview dialog.
+         webView.CoreWebView2.Print();
```

Comment:
Checking: we think or want most apps to show the browser UI rather than the system UI, which is why this is the default overload?